File: Shipping Project/Models/Order.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Shipping_Project.Models
{
    public class Order :BaseModel
    {
        public string CustomerName { get; set; }
        public string CustomerPhone1 { get; set; }
        public string? CustomerPhone2 { get; set; }
        public string VillageAndStreet { get; set; }
        public string Notes { get; set; }
        public double FinalCost { get; set; }
        public OrderState OrderState { get; set; }
        public OrderType OrderType { get; set; }
        public PaymentType PaymentType { get; set; }
        public double OrderPrice { get; set; }
        public int TotalWeight { get; set; }
        public bool IsDeleted { get; set; }=false;
        public bool ShippingToVillage { get; set; }
        public double AmountReceived { get; set; }
        //[ForeignKey("City")]
        //public int CityId { get; set; }
        //public virtual City? City { get; set; }
        [ForeignKey("ChargeType")]
        public int ChargeTypeId { get; set; }
        public virtual ChargeType? ChargeType { get; set; }
        public virtual List<Product>? Products { get; }=new List<Product>();
        [ForeignKey("RejectedReason")]
        public int? RrjectedResonId { get; set; } = null;
        public virtual RejectedReason? RejectedReason { get; set; }
        [ForeignKey("Branches")]
        public int BranchId { get; set; }
        public virtual Branches? Branches { get; set; }
        [ForeignKey("Merchent")]
        public int MerchentId { get; set; }
        public virtual Merchent? Merchent { get; set; }
        [ForeignKey("ShippigRepresentive")]
        public int ShippigRepresentiveId { get; set; }
        public virtual ShippigRepresentive? ShippigRepresentive { get; set; }

    }
    public enum OrderState
    {
        New,
        Pendding,
        DeliveredToTheRepresentative,
        Delivered,
        CannotBeReached,
        PostPoned,
        PartiallyDelivered,
        CanceledByCustomer,
        RejectedWithPayment,
        RejectedWithPartialPayment,
        RejectedWithoutPayment
    }
    public enum OrderType
    {
        DeliveryAtBranch,
        PickupFromTheMerchant
    }
    public enum PaymentType
    {
        CashOnDelivery,
        PaidInAdvance,
        ExchangeOrder,
    }
}

File: src/app/core/models/order.ts
export interface Order {
  id: number
  creationDate: string
  customerName: string
  customerPhone1: string
  customerPhone2?: string
  villageAndStreet: string
  notes?: string
  orderState: OrderState
  orderType: OrderType
  paymentType: PaymentType
  chargePrice: number
  orderPrice: number
  amountReceived: number
  totalWeight: number
  isDeleted: boolean
  isShippingToVillage: boolean
  cityName?: string
  chargeTypeName?: string
  branchName?: string
  merchantName?: string
  shippingRepresentativeName?: string
  products?: Product[]
}

export interface Product {
  id: number
  name: string
  weight: number
  quantity: number
  orderId: number
}

export interface CreateOrderDTO {
  customerName: string
  customerPhone1: string
  customerPhone2?: string
  villageAndStreet: string
  notes?: string
  orderPrice: number
  shippingToVillage: boolean
  cityId: number
  chargeTypeId: number
  branchId: number
  merchantId: string
  orderType: OrderType
  paymentType: PaymentType
  products: Product[]
}

export interface AssignOrderToDeliveryDTO {
  orderState: OrderState
  shippingRepresentativeId: string
}

export interface UpdateOrderStatusDTO {
  orderState: OrderState
  notes?: string
}

export interface OrderParams {
  pageIndex: number
  pageSize: number
  search?: string
  orderState?: OrderState
  cityId?: number
  branchId?: number
  merchantId?: string
  shippingRepresentativeId?: string
  
  fromDate?: string
  toDate?: string
  orderType?: OrderType
  paymentType?: PaymentType
  isDeleted?: boolean
  sort?: string
}

export interface PaginatedResponse<T> {
  pageSize: number
  pageIndex: number
  totalCount: number
  data: T[]
}

// String literals
export type OrderState =
  | "New"
  | "Pendding"
  | "DeliveredToTheRepresentative"
  | "Delivered"
  | "CannotBeReached"
  | "PostPoned"
  | "PartiallyDelivered"
  | "CanceledByCustomer"
  | "RejectedWithPayment"
  | "RejectedWithPartialPayment"
  | "RejectedWithoutPayment"

export type OrderType = "DeliveryAtBranch" | "PickupFromTheMerchant"
export type PaymentType = "CashOnDelivery" | "PaidInAdvance" | "ExchangeOrder"

// Static objects for reference
export const OrderStateValues = {
  New: "New" as OrderState,
  Pendding: "Pendding" as OrderState,
  DeliveredToTheRepresentative: "DeliveredToTheRepresentative" as OrderState,
  Delivered: "Delivered" as OrderState,
  CannotBeReached: "CannotBeReached" as OrderState,
  PostPoned: "PostPoned" as OrderState,
  PartiallyDelivered: "PartiallyDelivered" as OrderState,
  CanceledByCustomer: "CanceledByCustomer" as OrderState,
  RejectedWithPayment: "RejectedWithPayment" as OrderState,
  RejectedWithPartialPayment: "RejectedWithPartialPayment" as OrderState,
  RejectedWithoutPayment: "RejectedWithoutPayment" as OrderState,
}

export const OrderTypeValues = {
  DeliveryAtBranch: "DeliveryAtBranch" as OrderType,
  PickupFromTheMerchant: "PickupFromTheMerchant" as OrderType,
}

export const PaymentTypeValues = {
  CashOnDelivery: "CashOnDelivery" as PaymentType,
  PaidInAdvance: "PaidInAdvance" as PaymentType,
  ExchangeOrder: "ExchangeOrder" as PaymentType,
}

// For display in UI (e.g., dropdowns)
export const OrderStateOptions = Object.values(OrderStateValues).map((value) => ({
  value,
  label: value.replace(/([A-Z])/g, " $1").trim(),
}))

export const OrderTypeOptions = Object.values(OrderTypeValues).map((value) => ({
  value,
  label: value.replace(/([A-Z])/g, " $1").trim(),
}))

export const PaymentTypeOptions = Object.values(PaymentTypeValues).map((value) => ({
  value,
  label: value.replace(/([A-Z])/g, " $1").trim(),
}))
